### ISSUE TITLE
Document and update logic in controllers for handling the resources identified by either ID or Name.

### ISSUE THREAD

**Issue by Karthik-K-N** [kubernetes-sigs/cluster-api-provider-ibmcloud#1726]
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

Currently for any cloud resource for example [PowerVS workspace](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/fbbd4e6bec827bcd9dc5e4d9eab15823d403027f/api/v1beta2/ibmpowervsmachine_types.go#L141), TransitGateway, VPC user can specify it either via ID, Name and in some cases regular expressions and there are some assumptions we made and which needs to be documented
1. If user sets ID we expect it to be exist in cloud or lese fails the and update it via Event/condition on resource.
2. If user sets Name and the resource does not exist with given name we create it.

Along with documentation we need to revisit the controller code to ensure that this flow is achieved.

Few other things to note
1. For VPC subnets and loadbalancers we should not allow users to set duplicate names.
2. Document that if both name and id provided id will get precedence and update logic accordingly.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Comment by Karthik-K-N** [comment]
@Amulyam24 I think this is handled as part of your refactor PR right?

**Comment by Amulyam24** [comment]
yes, this has been handled by https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1937. It was referenced in the PR
Closing the issue.
/close
